File: Practice/InterviewBit/String/Longest Common Prefix.cpp
string Solution::longestCommonPrefix(vector<string> &A) {
    sort(A.begin(), A.end(), [](string a, string b){return a.length()<b.length();});
    
    string result=A[0];
    for(int i=1; i<A.size(); i++){
        int k=0;
        while(k<result.length() && result.substr(0, result.length()-k)!=A[i].substr(0, result.length()-k))   k++;
        
        result = result.substr(0, result.length()-k);
    }
    
    return result;
}

File: LeetCode/Longest-Common-Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String result = strs[0];
        for(int i=1; i<strs.length; i++){
            StringBuffer tempStr = new StringBuffer();

            for(int j=0; j<result.length(); j++){
                if(j<strs[i].length() && strs[i].charAt(j) == result.charAt(j)){
                    tempStr.append(result.charAt(j));
                }
                else{
                    break;
                }
            }

            result = tempStr.toString();
        }

        return result;
    }
}